src/main/java/com/kosherjava/zmanim/hebrewcalendar/JewishCalendar.java

```diff
@@ -617,11 +618,10 @@ public int getYomTovIndex() {
 					|| (!inIsrael && (day == 16 || day == 22))) {
 				return PESACH;
 			}
-			if (day >= 17 && day <= 20
-					|| (day == 16 && inIsrael)) {
+			if (day >= 17 && day <= 20 || day == 16) {
 				return CHOL_HAMOED_PESACH;
 			}
-			if ((day == 22 && inIsrael) || (day == 23 && !inIsrael)) {
+			if (day == 22 || day == 23 && !inIsrael) {
```

Comment:
We can remove the inIsrael check here, because it will never occur since PESACH will be returned above.